includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -2000,6 +2000,25 @@ public function save_vat_details( $vat_number, $name, $address ) {
 		return $response;
 	}
 
+	/**
+	 * Get if the merchant is eligible for Progressive Onboarding.
+	 *
+	 * @param array $business_info Business information.
+	 *
+	 * @return array HTTP response on success.
+	 *
+	 * @throws API_Exception - If not connected to server or request failed.
+	 */
+	public function get_onboarding_po_eligible( $business_info ) {
```

Comment:
This file is big and there are all REST API calls in here. I think in this PR I won't split it, but I guess it's a good idea for cooldown. Split it at least by APIs, like in the screenshot.

<img width="988" alt="image" src="https://user-images.githubusercontent.com/79862886/225688566-3fc20d60-38cf-454b-a711-d63c5f357452.png">

The idea is to create `WC_Payments_Accounts_API` that will contain only relevant API calls. And the main class `WC_Payments_API_Client` can use it.
Like 
```
public function add_tos_agreement( $source, $user_name ) {
		return $this->accounts_api->add_tos_agreement( $source, $user_name );
	}

```
@Automattic/transact-architecture what do you think about it? Maybe there is a discussion somewhere that I've missed?

Reply:
pinging @dechov as you are an architect assigned to Progressive Onboarding MVP project

Reply:
Apologies for the delay – my GH emails don't seem to distinguish pings from other subscribed activity well enough

Splitting it sounds great to me, though I'd be interested in hearing anyone else's thoughts on how to structure this. Maybe the main class doesn't even need methods for all of the calls, and some could be accessed via public properties like `$this->api_client->accounts-> add_tos_agreement( $source, $user_name );` ?

Reply:
@ismaeldcom WDYT?

Reply:
I agree, it's good to find code to improve while working on a PR, but we should try to not deviate that much from the initial estimation to include them. A cooldown issue for this sounds good 🙂

Reply:
Issue created https://github.com/Automattic/woocommerce-payments/issues/5856